mountpoint-s3-client/tests/head_bucket.rs

```diff
@@ -41,3 +41,16 @@ async fn test_head_bucket_forbidden() {
         Err(ObjectClientError::ServiceError(HeadBucketError::PermissionDenied(_)))
     ));
 }
+
+#[tokio::test]
+async fn test_head_bucket_not_found() {
+    let client = get_test_client();
+    let bucket = "test-nosuchbucket-s3alias";
```

Comment:
Tried to pick a bucket name that won't ever exist. `-s3alias` is a reserved suffix for Access Point bucket-based aliasing.